### PR TITLE
Remove code hash from EVM genesis config's

### LINKED ISSUE #356: EVM: enhance genesis configs
TODOs:

- [x] Calculate `code_hash` (don't read from genesis file)
- [x] Set nonce to 1 for predeploy contracts (and remove from genesis configs)

### PR DESCRIPTION
**Pull request by otaliptus**
# Description
Fixes the first part of #356.

### CODE REVIEW

File: crates/evm/src/genesis.rs [hunk 1]
```diff
@@ -255,12 +292,13 @@ mod tests {
...
         let address = Address::from_str("0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266").unwrap();
+        let code = Bytes::from_hex("0x60606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063a223e05d1461006a578063").unwrap();
         let config = EvmConfig {
             data: vec![AccountData {
                 address,
                 balance: AccountData::balance(u64::MAX),
-                code_hash: AccountData::empty_code(),
-                code: Bytes::from_hex("0x60606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063a223e05d1461006a578063").unwrap(),
+                code_hash: keccak256(&code),
+                code: code,
```
Comment (otaliptus) [inline crates/evm/src/genesis.rs:301]: Which one is correct here?

File: crates/evm/src/tests/genesis_tests.rs [hunk 2]
```diff
@@ -34,7 +34,9 @@ lazy_static! {
             address:Address::from([2u8; 20]),
             balance: U256::checked_mul(U256::from(1000),
             U256::pow(U256::from(10), U256::from(18))).unwrap(), // 1000 ETH,
-            code_hash: B256::from_slice(&[2u8; 32]), code: Bytes::from_hex("60606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063a223e05d1461006a578063").unwrap(), storage: {
+            code_hash: B256::from_hex("0x4e8ee9adb469b245e3a5a8e58e9b733aaa857a9dce1982257531db8a2700aabf").unwrap(),
+            code: Bytes::from_hex("0x60606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063a223e05d1461006a578063").unwrap(),
```
Comment (otaliptus) [inline crates/evm/src/tests/genesis_tests.rs:38]: Same

File: crates/evm/src/tests/genesis_tests.rs [hunk 3]
```diff
@@ -60,10 +62,10 @@ lazy_static! {
     };
 
     pub(crate) static ref GENESIS_HASH: B256 = B256::from(hex!(
-        "21d33ac980d24d19397167ce9b4e9fb072d7942f22c2451e8d49c9b1c3bc1ff1"
+        "d3d0598f798aeac0eb991a29a1ec2e79e6d3caf76393c787a564d89233342df8"
     ));
     pub(crate) static ref GENESIS_STATE_ROOT: B256 = B256::from(hex!(
-        "cf60bd2197418b787dc8a3e1befea15f15546bc60a12075fe06da5ee9a284e95"
+        "050d41ea04851c5800ba402d42d7a3777ca025feda7c1bd6a272ff9710a4501a"
```
Comment (otaliptus) [inline crates/evm/src/tests/genesis_tests.rs:68]: These were supposed to change, I guess?

File: bin/test-data/genesis/demo-tests/celestia/evm.json
Comment (eyusufatik) [inline bin/test-data/genesis/demo-tests/celestia/evm.json:1]: Delete this

**Review by eyusufatik — CHANGES_REQUESTED** [review]
Are you planning to do the nonce thing in another pr?

**Comment by otaliptus** [comment]
Another

**Review by eyusufatik — CHANGES_REQUESTED** [review]

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**